contracts/Aavegotchi/facets/AavegotchiLendingFacet.sol

```diff
@@ -0,0 +1,291 @@
+    ///@param _rentalId The identifier of the rental to agree
+    function agreeAavegotchiRental(
+        uint256 _rentalId,
+        uint256 _erc721TokenId,
+        uint256 _amountPerDay,
+        uint256 _period,
+        uint256[3] memory _revenueSplit
+    ) external {
+        AavegotchiRental storage rental = s.aavegotchiRentals[_rentalId];
+        require(rental.timeCreated != 0, "AavegotchiLending: rental not found");
+        require(rental.timeAgreed == 0, "AavegotchiLending: rental already agreed");
+        require(rental.canceled == false, "AavegotchiLending: rental canceled");
+        require(rental.erc721TokenId == _erc721TokenId, "AavegotchiLending: Invalid token id");
+        require(rental.amountPerDay == _amountPerDay, "AavegotchiLending: Invalid amount per day");
+        require(rental.period == _period, "AavegotchiLending: Invalid rental period");
+        for (uint256 i; i < 3; i++) {
+            require(rental.revenueSplit[i] == _revenueSplit[i], "AavegotchiLending: Invalid revenue split");
+        }
+        address renter = LibMeta.msgSender();
+        address originalOwner = rental.originalOwner;
+        require(originalOwner != renter, "AavegotchiLending: renter can't be original owner");
+        require(rental.whitelistId == 0 || s.isWhitelisted[rental.whitelistId][renter], "AavegotchiLending: Not whitelisted address");
+
+        if (rental.amountPerDay > 0) {
+            uint256 transferAmount = rental.amountPerDay * rental.period;
+            require(IERC20(s.ghstContract).balanceOf(renter) >= transferAmount, "AavegotchiLending: not enough GHST");
+            LibERC20.transferFrom(s.ghstContract, renter, originalOwner, transferAmount);
+        }
+
+        rental.renter = renter;
+        rental.timeAgreed = block.timestamp;
+
+        uint256 tokenId = rental.erc721TokenId;
+        s.lentTokenIdIndexes[originalOwner][tokenId] = s.lentTokenIds[originalOwner].length;
+        s.lentTokenIds[originalOwner].push(tokenId);
+
+        if (rental.erc721TokenAddress == address(this)) {
+            LibAavegotchi.transfer(originalOwner, renter, tokenId);
+        } else {
+            // External contracts
+            IERC721(rental.erc721TokenAddress).safeTransferFrom(originalOwner, renter, tokenId);
+        }
+
+        // set original owner as pet operator
+        s.petOperators[renter][originalOwner] = true;
+
+        emit ERC721ExecutedRental(
+            _rentalId,
+            originalOwner,
+            renter,
+            rental.erc721TokenAddress,
+            tokenId,
+            rental.amountPerDay,
+            rental.period,
+            block.timestamp
+        );
+    }
+
+    ///@notice Allow a original owner to claim revenue from the rental
+    ///@dev Will throw if the NFT has not been lent or if the rental has been canceled already
+    ///@param _rentalId The identifier of the rental to claim
+    function claimAavegotchiRental(uint256 _rentalId) external {
```

Comment:
Then rentalId for an Aavegotchi could change.   It seems to me that it would be easier and more stable for users or UIs if this function took a tokenID as an argument rather than a rentalId.  Is there an advantage or benefit to using a rentalId instead of a token ID?

Reply:
I will use both rental id and aavegotchi id.

Reply:
To be clear, I'm not concerned about block reorganizations here.    I'm just thinking that it might be easier for a person or app  to remember or retrieve what Aavegotchi tokenID to use to claim,  rather than a rentalID.